### DISCUSSION TITLE
Trying to start helping

### DISCUSSION THREAD

**Discussion by spirann** [joleys/niko-home-control-II#185] [Q&A] [answered]
Hello,

I'd like to help the project, so I've implemented the water central meter
you can see it here : https://github.com/spirann/niko-home-control-II/commit/cfd50d8d768d41e2faa4a114de5321804f133dda

but when I restart home assistant I got this in the logs : 

2024-10-02 16:17:36.512 DEBUG (Thread-3 (_thread_main)) [custom_components.nhc2.nhccoco.coco] Added device f8e2810d-0974-4222-bd7f-2e5996dcec36 of class CocoLightAction
**2024-10-02 16:17:36.512 DEBUG (Thread-3 (_thread_main)) [custom_components.nhc2.nhccoco.coco] Skipping 7f9bc0ef-b2d2-41c3-a9af-9ce6e2a4aaed of CocoWaterCentralmeter**
2024-10-02 16:17:36.512 DEBUG (Thread-3 (_thread_main)) [custom_components.nhc2.nhccoco.coco] Added device c77c46f4-09b2-4ffd-bb6c-07ef8c770345 of class CocoSwitchedGenericAction

Why is it skipping my watercentralmeters ?

**Comment by tijsverkoyen** [comment]
See https://github.com/joleys/niko-home-control-II/blob/f66109b6e330e751dbd86989236ced1ac90a1b84/custom_components/nhc2/nhccoco/coco.py#L406
To my knowledge the `CocoWaterCentralmeter` does not expose any useful properties. Unless there is an update?

**Reply by spirann** [reply]
It should contain the the water volume. I'm trying to get my consommation into HA
I've implemented 
`@property
    def water_volume(self) -> float:
        return to_float_or_none(self.extract_property_value(PROPERTY_WATER_VOLUME))`

based on the received device list:
`{'Uuid': 'd60f1fff-05ff-48ff-90ff-b2e7bb93a5ff', 'Identifier': 'ed36adff-a5ff-48ff-bfff-86fba30ef6ff', 'Model': 'water', 'Online': 'True', 'Name': 'Boiler', 'Technology': 'nikohomecontrol', 'Type': 'centralmeter', 'Traits': [{'Channel': '0'}, {'MacAddress': '00200FAFF'}], 'Parameters': [{'LocationId': '0ee306ff-30ff-42ff-a4ff-89fec29dfcff'}, {'Segment': 'Subsegment'}, {'PulsesPerUnit': '1000'}, {'LocationIcon': 'cold-storage-room'}, {'Flow': 'Consumer'}, {'ShortName': 'Boiler'}, {'LocationName': 'Annexe'}], 'Properties': [{'WaterVolume': ''}], 'PropertyDefinitions': [{'WaterVolume': {'HasStatus': 'false', 'CanControl': 'false', 'Description': 'Range(0.0000,3.2512,0.0001)'}}]}`

Why should it not contains any useful properties ? Water Volume stays empty ?I though it will be refreshed with real value.
 Why exposing it via api if not to be exploited ?

**Reply by tijsverkoyen** [reply, marked as answer]
Aha in previous versions these propers were not exposed (to my knowledge).
In that case you will need to remove this ignore class: https://github.com/joleys/niko-home-control-II/blob/f66109b6e330e751dbd86989236ced1ac90a1b84/custom_components/nhc2/nhccoco/coco.py#L406

**Reply by spirann** [reply]
Thanks to you, I have fixed my implementation. But like you thought, in the Niko documentation it's described as not implemented and the value stays empty :  `'Properties':` [{'WaterVolume': ''}]`. 
But it was a good training, I just need to wait for Niko to insert the value....
